### ISSUE TITLE
Populate row cache using compacting sstable reader

### ISSUE THREAD

**Issue by tgrabiec** [scylladb/scylla#3568]
Currently, the sstable reader underlying the cache is not compacting, so cache may be populated with expired or deleted elements, which wastes memory and makes reads slower, until relevant sstables get compacted.

**Comment by tgrabiec** [comment]
@denesb We can now fix this since we have a compacting reader.

**Comment by denesb** [comment]
We need some further changes to expose https://github.com/scylladb/scylla/blob/14de126ff83d8322090e4190273aecf06ce64b31/sstables/compaction.cc#L76. Or we could do this as part of adding the first user of the compacting reader, be that cache or streaming.

**Comment by tgrabiec** [comment]
I don't understand the problem.

Ignoring #1745, isn't `get_max_purgeable_timestamp()` equal to `api::max_timestamp` for cache sstable reader? We read all sstables, so can drop whatever expired.  To account for #1745, we'd have to account for memtables.

**Comment by denesb** [comment]
Good point, we are opening all relevant sstables, so we don't even need that, we can just use `api::min_timestamp` and purge all tombstones.

**Comment by avikivity** [comment]
Performance, not backporting.
